src/block.rs

```diff
@@ -1,28 +1,31 @@
+#[cfg(target_arch = "x86_64")]
+type BlockAddr = u64;
```

Comment:
The commit message for this commit isn't quite accurate.

We use a `u64` not because we only work on x86_64, but because [libipt (which only works on x86_64) uses `uint64_t` to express block addresses](https://github.com/intel/libipt/blob/c848a85c3104e2f5780741f85de5c9e65476ece2/libipt/include/intel-pt.h.in#L2381-L2389).

So this commit is really about supporting different decoders, not different architectures.

Would it make more sense to use `usize`, as in most chips, the instruction pointer is exactly wide enough to store a pointer?

Reply:
(and i'm assuming that we don't want to cater for decoding a trace for a different platform)

Reply:
A question which I haven't fully worked out the answer to is: how generic is `hwtracer` supposed to be? At the moment it supports various backends, of which libipt/PT is but one, so having libipt's notion of address being the (possibly generic) `Block` idea of address seems odd. But an alternative thought is that this library should actually be called something like "pttracer", at which point that seems less of a concern.

Reply:
I see what you mean. I only implemented stuff at the block-level, as that's the only granularity we care about right now.

(libipt also has a decoder that works at instruction-level granularity)

Reply:
I think leaving it generic-ish for the time being is OK: that means we can later easily make it a bit more or less generic as we think is best.